packages/common/src/components/form/TimeRangePicker.tsx

```diff
@@ -0,0 +1,162 @@
+  }
+`;
+
+const StartBeforeEndError = styled.span`
+  grid-column: span 2;
+  color: var(--color-error);
+  margin-top: var(--spacing-2-xs);
+`;
+
+/*
+ *  @brief A component for selecting a time range, checks for whether the end time is before the begin time
+ *  @param {Control} control - the parent form's `control` object (required)
+ *  @param {begin: Path<T>, end: Path<T>} name - the name objects for the time values (required)
+ *  @param {begin?: boolean, end?: boolean} [name] - flags for whether the begin and/or end times are required (optional)
+ *  @param {begin?: string, end?: string} [label] - the label text for the begin and/or end times (optional)
+ *  @param {begin?: boolean, end?: boolean} [placeholder] - a function to execute upon clicking the button (optional)
+ *  @param {begin?: boolean, end?: boolean} [clearable] -
+ *  @returns {JSX.Element} Two `<Select>` components for selecting a begin and end time, populated by options from
+ *  00:00 to 23:59 in 30min intervals. If the end time is before the begin time, an error message is displayed (the
+ *  error is internal, and does not trigger invaliditiy of parent form). Returned elements don't have a wrapping
+ *  element, so wrap the component in the parent form if needed.
+ */
+const TimeRangePicker = <T extends FieldValues>({
+  control,
+  name,
+  required,
+  label,
+  placeholder,
+  clearable,
+}: TimeRangePickerProps<T>): JSX.Element => {
+  const { field: beginField, fieldState: beginFieldState } = useController({
+    control,
+    name: name?.begin,
+    rules: { required: required?.begin },
+  });
+  const { field: endField, fieldState: endFieldState } = useController({
+    control,
+    name: name?.end,
+    rules: { required: required?.end },
+  });
+  const { t } = useTranslation();
+
+  // Return the option with the given value as label
+  const getSelectedOption = (
+    optionValue: number | null,
+    optionList: OptionType[]
+  ): OptionType | null => {
+    return optionList.find((o) => o.label === optionValue) ?? null;
+  };
+  const populateTimes = (
+    populateTimesProps?: PopulateTimesProps
+  ): OptionType[] => {
+    const beginHour = populateTimesProps?.beginHour ?? 0;
+    const endHour = populateTimesProps?.endHour ?? 23.98;
+    const interval = populateTimesProps?.intervalMinutes ?? 30;
+    const times: OptionType[] = [];
+    let hour = beginHour ?? 0;
+    let minute = beginHour % 1 ? (beginHour % 1) * 60 : 0;
+
+    while (hour < endHour) {
+      times.push({
+        label: `${hour.toString().padStart(2, "0")}:${minute
+          .toString()
+          .padStart(2, "0")}`,
+        value: hour + minute / 60,
+      });
+      minute += interval;
+      // Reset the minute counter, and increment the hour counter if necessary
+      if (minute >= 60) {
+        minute = 0;
+        hour += 1;
+      }
+    }
+    // we need to add the minute times to the beginning of the duration options
+    return times as OptionType[];
```

Comment:
No, probably not. I think I was first casting it, before specifying it in the function declaration and this was left behind (..and didn't give any errors 😇). I'll remove the extraneous typecasting - but just out of curiosity, is there some harm in typecasting a variable to the same type that it infers?

Reply:
Type casts remove type checks, technically TS 5+ doesn't allow casting unmatched types without going through unknown but it might (depends on settings and types).
So `undefined as number` might be valid code and it's gonna break on runtime. In this case if the array might be undefined the cast would probably be valid, but would crash on runtime.